### PR TITLE
[dagster-gcp-pyspark] Add BigQuery PySpark type handler

### PR DESCRIPTION
**Pull request by jamiedemaria**
### Summary & Motivation
Adds a PySpark type handler for the BigQuery IO manager

API docs preview https://dagster-git-jamie-bigquery-pyspark-elementl.vercel.app/_apidocs/libraries/dagster-gcp-pyspark

### How I Tested These Changes
bk + new unit tests

### CODE REVIEW

File: python_modules/libraries/dagster-gcp-pyspark/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+include README.md
+include LICENSE
+recursive-include dagster_gcp_pyspark *.py
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-gcp-pyspark/MANIFEST.in:3]: remove and replace with the py.typed line

File: docs/content/_apidocs.mdx [hunk 2]
```diff
@@ -88,6 +88,10 @@ Dagster also provides a growing set of optional add-on libraries to integrate wi
 
 - [GCP](/\_apidocs/libraries/dagster-gcp) (`dagster-gcp`) Dagster integrations for working with Google Cloud Platform resources.
 
+- [GCP+Pandas](/\_apidocs/libraries/dagster-gcp-pandas) (`dagster-gcp-pandas`) Dagster integrations for working with Google Cloud Platform resources with Pandas DataFrames.
+
+- [GCP+PySpark](/\_apidocs/libraries/dagster-gcp-pyspark) (`dagster-gcp-pyspark`) Dagster integrations for working with Google Cloud Platform resources with PySpark DataFrames.
```
Comment (erinkcochran87) [inline docs/content/_apidocs.mdx:93]: On the API reference page itself, one of the first headings says `BigQuery` but the description here doesn't mention that at all. Does this work with all GCP resources or just BigQuery?

**Review by erinkcochran87 — COMMENTED** [review]
Also left some comments on the Vercel preview, but given that it looks like there were some issues with the deployment, I'm not sure if it's totally accurate. 🤷‍♀️

File: docs/content/_apidocs.mdx (hunk 2 above)
Comment (jamiedemaria) [reply docs/content/_apidocs.mdx:93]: right now there's only bigquery support, but in the future this library could have things for other GCP resources

Comment (erinkcochran87) [reply docs/content/_apidocs.mdx:93]: :+1: D'you think it's worth mentioning in the description here? I know some folks have asked about support for BQ specifically, so my thinking was that would make it easier to find if they're skimming

File: python_modules/libraries/dagster-gcp-pyspark/dagster_gcp_pyspark/bigquery/bigquery_pyspark_type_handler.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        defs = Definitions(
+            assets=[my_table],
+            resources={
+                "io_manager": bigquery_pyspark_io_manager.configured({
+                    "project" : {"env": "GCP_PROJECT"}
+                })
+            }
+        )
+
+    You can tell Dagster in which dataset to create tables by setting the "dataset" configuration value.
```
Comment (sryza) [inline python_modules/libraries/dagster-gcp-pyspark/dagster_gcp_pyspark/bigquery/bigquery_pyspark_type_handler.py:124]: Should this be 'the "schema" metadata value'?

File: python_modules/libraries/dagster-gcp-pyspark/setup.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        "Programming Language :: Python :: 3.10",
+        "License :: OSI Approved :: Apache Software License",
+        "Operating System :: OS Independent",
+    ],
+    packages=find_packages(exclude=["dagster_gcp_pyspark_tests*"]),
+    install_requires=[
+        f"dagster{pin}",
+        f"dagster-gcp{pin}",
+        "pyspark",
+        "requests",
```
Comment (sryza) [inline python_modules/libraries/dagster-gcp-pyspark/setup.py:42]: Why requests?

**Review by sryza — APPROVED** [review]
Two small comments - otherwise, this is awesome

File: python_modules/libraries/dagster-gcp-pyspark/dagster_gcp_pyspark/bigquery/bigquery_pyspark_type_handler.py (hunk 3 above)
Comment (jamiedemaria) [reply python_modules/libraries/dagster-gcp-pyspark/dagster_gcp_pyspark/bigquery/bigquery_pyspark_type_handler.py:124]: it's `dataset` when it's configured on the io manager itself. The only time it's `schema` is for output metadata on ops

**Comment by jamiedemaria** [comment]
@sryza @benpankow i made a couple changes over the past few days to get tests working, if you'd like to review again!

**Pull request merged**